lib/rubocop/cop/rails/orm_abstraction.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # Avoid breaking ORM abstraction.
+      # @example
+      #
+      #   # bad
+      #   where(foo_id: foo.id)
+      #
+      #   # bad
+      #   create(foo_id: foo.id)
+      #
+      #   # good
+      #   where(foo: foo)
+      #
+      #   # good
+      #   create(foo: foo)
+
+      class ORMAbstraction < Cop
+        def_node_matcher :bad_pair?, <<-PATTERN
+          (pair {str_type? sym_type? dstr_type?} (send !nil? {:id :uuid}))
+        PATTERN
+
+        def on_pair(node)
+          return unless bad_pair?(node)
+          return if !node.key.source.match(BAD_RXP)
+
+          corrector = Corrector.new(processed_source.buffer)
+          replacement = prefered_pair(node, corrector)
+            .process_node(node.loc.expression)
+
+          add_offense(
+            node,
+            location: node.loc.expression,
+            message: "prefer `#{replacement}`."
+          )
+        end
+
+        BAD_RXP = /_(id|uuid)(?="|'|$)/
+
+        def autocorrect(node)
+          lambda do |corrector|
+            prefered_pair(node, corrector)
+          end
+        end
+
+        def prefered_pair(bad_pair, corrector)
+          key, val = bad_pair.to_a
+          prefered_key(key, corrector)
+          prefered_val(val, corrector)
+        end
+
+        def prefered_key(key, corrector)
+          corrector.replace(key.loc.expression, key.source.sub(BAD_RXP, ''))
+        end
+
+        def prefered_val(val, corrector)
+          corrector.replace(val.loc.expression, val.receiver.source)
+        end
+      end
+    end
+  end
+end
+
+class Parser::Source::TreeRewriter
```

Comment:
This is going to be temporary. Either I think of a way to do this without changing TreRewriter or we just open a PR on TreeRewriter.